### PR TITLE
validate whitelist vs. validate pairs

### PR DESCRIPTION
**Pull request by iuvbio**
## Summary
Loop over whitelist only instead of all market pairs in `ipairlist._validate_whitelist` and (potentially) replace `exchange.validate_pairs`.

Solve the issue: #1622

## Quick changelog

- change `ipairlist._validate_whitelist` to loop over whitelist instead of markets. Also combine stake currency check with main validation.
- change `exchange.validate_pairs` to check only whether pair is in loaded markets

## What's new?
I've mainly combined the conditions in `_validate_whitelist` which brings a significant performance improvement as can be seen below:
image

I haven't removed `exchange.validate_pairs` yet, as I'd like to get your input.. As I wrote in #1622, that check is basically unnecessary, since the whitelist gets validated by `ipairlist` anyways.. However, it's only executed once when the exchange is initialised. So not too much damage and the whitelist gets validated earlier. Could also import _validate_whitelist and call it there, but that seems like overkill.. imo, it's probably fine to take it out. What do you think?

## TBD
- [x] Adapt tests

### CODE REVIEW

**Comment by mishaker** [comment]
@iuvbio tell us when we can review it :) we usually don't do review when there is [WIP] on the title and when are still TODOs :)

**Comment by iuvbio** [comment]
Should be good to go, I haven't removed `validate_pairs`.

File: freqtrade/pairlist/IPairList.py [hunk 1]
```diff
@@ -60,34 +60,29 @@ def refresh_pairlist(self) -> None:
...
-            known_pairs.add(pair)
-            # Market is not active
+            # Check if market is active
+            market = markets[pair]
             if not market['active']:
-                sanitized_whitelist.remove(pair)
                 logger.info(
-                    'Ignoring %s from whitelist. Market is not active.',
-                    pair
+                    f"Ignoring {pair} from whitelist. Market is not active."
```
Comment (xmatthias) [inline freqtrade/pairlist/IPairList.py]: I think we should try to move log-statements to 1 line if it doesn't upset flake8 ...

File: freqtrade/tests/pairlist/test_pairlist.py [hunk 2]
```diff
@@ -107,40 +107,27 @@ def test_VolumePairList_refresh_empty(mocker, markets_empty, whitelist_conf):
...
 
 
-def test_VolumePairList_whitelist_gen(mocker, whitelist_conf, markets, tickers) -> None:
+@pytest.mark.parametrize("precision_filter,base_currency,key,whitelist_result", [
+    (False, "BTC", "quoteVolume", ['ETH/BTC', 'TKN/BTC', 'BTT/BTC']),
+    (False, "BTC", "bidVolume", ['BTT/BTC', 'TKN/BTC', 'ETH/BTC']),
+    (False, "USDT", "quoteVolume", ['ETH/USDT', 'LTC/USDT']),
+    (False, "ETH", "quoteVolume", []),  # this replaces tests that were removed from test_exchange
+    (True, "BTC", "quoteVolume", ["ETH/BTC", "TKN/BTC"]),
+    (True, "BTC", "bidVolume", ["TKN/BTC", "ETH/BTC"])
```
Comment (xmatthias) [inline freqtrade/tests/pairlist/test_pairlist.py]: Great test refactorings! :+1:

**Pull request merged**